### PR TITLE
Update TS documentation for 0.12.0

### PR DESCRIPTION
**Pull request by kipz**
***Please merge only when 0.12.0 is released***

- Use proper code highlighting keywords
- Update for 0.12.0
- Add editors/samples in to repo itself - we could link to these too?

### CODE REVIEW

File: docs/quick-starts/rug-typescript.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ As a starting point, simply add a [standard `package.json` file](https://docs.np
 ```json
 {
   "dependencies": {
-    "@atomist/rug": "0.10.0"
+    "@atomist/rug": "0.11.0"
```
Comment (ddgenome) [inline docs/quick-starts/rug-typescript.md]: 0.12.0?

File: docs/quick-starts/rug-typescript.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ Once you have these two installed on your machine you should be able to successf
 
 ```shell
 $ npm install
-@russmiles/scattered-rugs@0.8.0 ...scattered-rugs/.atomist
+@yourusername/scattered-rugs@0.8.0 ...scattered-rugs/.atomist
 └── @atomist/rug@0.10.0
```
Comment (ddgenome) [inline docs/quick-starts/rug-typescript.md]: Perhaps default version in sample should be 0.1.0 to agree with other content and be less surprising.

Comment (ddgenome) [inline docs/quick-starts/rug-typescript.md]: Should the rug version here be 0.12.0 to agree with above?

Comment (Lawouach) [inline docs/quick-starts/rug-typescript.md]: the version here should match the one in the dependencies above, shouldn't it?

File: docs/quick-starts/rug-typescript.md (hunk 1 above)
Comment (Lawouach) [inline docs/quick-starts/rug-typescript.md]: should this be 0.12?

File: docs/quick-starts/rug-typescript.md [hunk 3]
```diff
@@ -89,51 +89,46 @@ Known operations are []
...
 
--   Importing the TypeScript types for working with Rug.
--   Declaring a single, mandatory parameter `description` to this editor.
--   Tagging and exporting a `ProjectEditor` implementation that contains the `edit` function implementation for the project-editing logic.
+-   Importing the TypeScript typings for working with Rug.
+-   Declaring the editor using the `@Editor` decorator, passing in its name and description
+-   Tagging the editor using the `@Tag` decorator to aid discoverability
+-   Implementing the `EditProject` interface to enforce the `edit` function signature
+-   Declaring any require parameters using the `@Parameter` decorator
+   - The value of the field is used as the default value if no value is supplied during invocation
```
Comment (ddgenome) [inline docs/quick-starts/rug-typescript.md]: If you want this to be a subitem, it is safest and most portable if you indent four space and have the content start at the eighth character. That said, if there is only one subitem, it is best practice to roll it into the parent.

**Comment by kipz** [comment]
The files under .atomist are the samples. I'm hoping that they can be linked to and/or included directly at some point. Being part of an actual project is super useful for correctness.

File: docs/quick-starts/rug-typescript.md (hunk 3 above)
Comment (kipz) [inline docs/quick-starts/rug-typescript.md]: Odd that this is still showing. It's fixed in the commit!

**Comment by Lawouach** [comment]
You might want to look at http://facelessuser.github.io/pymdown-extensions/extensions/snippets/ to include those project files into the doc. I haven't tried yet but hopeful.

**Comment by Lawouach** [comment]
We should support that extension out of the box.

**Pull request merged**